GoogleForCreators/web-stories-wp
# Carousel: Refactor to adhere to new design

## Summary

This is a complete overhaul of the carousel components (code-wise) as well as adding a new structure for it.

I decided to update the thumbnail highlight to the new design (with a full thin border rather than just a fat top border) as the old design looked weird with the new changes.

## Relevant Technical Choices



## To-do

* [x] Refactor carousel
* [x] Change thumbnail sizing and layout
* [x] Retire old compact indicator
* [x] Change scroll arrow behavior
* [x] Change carousel menu layout
* [x] Update thumbnail highlight to new design
* [x] Make sure tests still run

## User-facing changes



## Testing Instructions

* Make sure thumbnail carousel, grid view and keyboard shortcuts still work as intended.

---



Fixes #5692
Fixes #5077
